### ISSUE TITLE
co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'PatternTokenizer.flags'

### ISSUE THREAD

**Issue by grib0** [elastic/elasticsearch-java#418]
### Java API client version

7.17.2

### Java version

17

### Elasticsearch Version

7.17.0

### Problem description

Using the java client api, I'm unable to create an index with a this mapping

```{
    "settings": {
        "analysis": {
            "tokenizer": {
                "espaceTiretTokenizer": {
                    "type": "pattern",
                    "pattern": "[\\s,/\\\\-]+"
                }
            }
        }
    }
}
```
        
```   
elasticsearchClient.indices()
            .create(c -> c.index(adresseIndex)
            .withJson("settings loaded here")

```
I got : 
`co.elastic.clients.util.MissingRequiredPropertyException: Missing required property 'PatternTokenizer.flags'`       
     
 Trying to add
`ApiTypeHelper.DANGEROUS_disableRequiredPropertiesCheck(true);
` as explained in the doc https://www.elastic.co/guide/en/elasticsearch/client/java-api-client/7.17/missing-required-property.html results in 
`java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "co.elastic.clients.util.ApiTypeHelper.requireNonNull(Object, Object, String)" is null`

The workaround is to add the 2 properties flags and group but this properties should be optional

**Comment by l-trotta** [comment]
Hello and thanks for the report! This is an [API specification](https://github.com/elastic/elasticsearch-specification) issue, and you are right, those properties should be optional, in fact they are in the most recent version of the 8.X client. We will apply the fix to 7.17 too, then the java client will be updated to solve this issue.

**Comment by l-trotta** [comment]
fixed with #759 , the issue will be solved in the next release.
